protocol/src/main/kotlin/net/rsprot/protocol/metrics/impl/GenericNetworkTrafficHandler.kt

```diff
@@ -0,0 +1,103 @@
+package net.rsprot.protocol.metrics.impl
+
+import net.rsprot.protocol.metrics.NetworkTrafficHandler
+import net.rsprot.protocol.metrics.channel.impl.GameChannelTrafficHandler
+import net.rsprot.protocol.metrics.channel.impl.Js5ChannelTrafficHandler
+import net.rsprot.protocol.metrics.channel.impl.LoginChannelTrafficHandler
+import net.rsprot.protocol.metrics.lock.TrafficHandlerLock
+import net.rsprot.protocol.metrics.snapshots.NetworkTrafficSnapshot
+import net.rsprot.protocol.metrics.snapshots.impl.GenericNetworkTrafficSnapshot
+import java.net.InetAddress
+import java.time.LocalDateTime
+import java.util.Queue
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentLinkedQueue
+import java.util.concurrent.atomic.AtomicInteger
+
+public class GenericNetworkTrafficHandler<LoginBlock>(
+    private val lock: TrafficHandlerLock,
+    override var loginChannelTrafficHandler: LoginChannelTrafficHandler,
+    override var js5ChannelTrafficHandler: Js5ChannelTrafficHandler,
+    override var gameChannelTrafficHandler: GameChannelTrafficHandler,
```

Comment:
seems to me like these can be `val`s unless i'm missing some mutation somewhere

Reply:
You are correct, yeah. I initially intended to swap out the instances of that whenever a snapshot was created, but that did not quite work out, need to reset internally instead. I've changed it in a future commit.